src/main/java/com/desiato/puresynth/services/AuthenticationService.java

```diff
@@ -0,0 +1,91 @@
+package com.desiato.puresynth.services;
+
+import com.desiato.puresynth.dtos.AuthenticationRequestDTO;
+import com.desiato.puresynth.dtos.PureSynthToken;
+import com.desiato.puresynth.models.CustomUserDetails;
+import com.desiato.puresynth.models.Session;
+import com.desiato.puresynth.models.User;
+import com.desiato.puresynth.repositories.SessionRepository;
+import com.desiato.puresynth.repositories.UserRepository;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Service
+public class AuthenticationService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private SessionRepository sessionRepository;
+
+    private static final Logger logger = LoggerFactory.getLogger(AuthenticationService.class);
```

Comment:
If you're using Lombok, you could use https://projectlombok.org/api/lombok/extern/slf4j/Slf4j.

Reply:
that's great! I did not know that!